docs/topics/collection-grouping.md

```diff
@@ -2,22 +2,33 @@
 
 The Kotlin standard library provides extension functions for grouping collection elements.
 The basic function [`groupBy()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/group-by.html) takes a
-lambda function and returns a `Map`. In this map, each key is the lambda result and the corresponding value is the `List`
+lambda function and returns a `Map`. In this map, each key is the lambda result, and the corresponding value is the `List`
 of elements on which this result is returned. This function can be used, for example, to group a list of `String`s by
 their first letter. 
 
 You can also call `groupBy()` with a second lambda argument – a value transformation function.
 In the result map of `groupBy()` with two lambdas, the keys produced by `keySelector` function are mapped to the results
 of the value transformation function instead of the original elements.
 
-```kotlin
+This example illustrates using the `for` operator to iterate through the groups created by the `groupBy()` function on the resulting `Map`:
 
+```kotlin
 fun main() {
 //sampleStart
     val numbers = listOf("one", "two", "three", "four", "five")
 
-    println(numbers.groupBy { it.first().uppercase() })
-    println(numbers.groupBy(keySelector = { it.first() }, valueTransform = { it.uppercase() }))
+    // Groups the strings by their first letter using groupBy
+    val groupedByFirstLetter = numbers.groupBy { it.first().uppercase() }
+    println(groupedByFirstLetter)
```

Comment:
From what I remember, we add a comment line with the output right after the print line for those that cant' run the code snippets.

Reply:
looked a bit too busy for me with the output included as well, but sure - it cannot hurt 👍